lib/rules/indent.js

```diff
@@ -379,12 +379,17 @@ module.exports = {
          * if not present then return null
          * @param {ASTNode} node node to examine
          * @param {string} type type that is being looked for
+         * @param {string} stopAt end point for the evaluating code
          * @returns {ASTNode|void} if found then node otherwise null
          */
-        function getParentNodeByType(node, type) {
+        function getParentNodeByType(node, type, stopAt) {
             let parent = node.parent;
 
-            while (parent.type !== type && parent.type !== "Program") {
+            if (!stopAt) {
```

Comment:
This block might not be necessary since we changed line 392 to check explicitly against Program, see if you can remove this?